MH-12948, Directly read XACML files

lkiesow opened the pull request:
The XACML authorization service passed configuration files to a JBoss
library to determine if a specific action is allowed for a user given an
XACML file. The configuration itself contained file locations specifying
where the library could load the XACML file. This caused Opencast to
repeatedly load the XACML attachments into the workspace, cleaning them
up again afterwards.

This patch now allows Opencast to directly read those files from the
working file repository by using Opencast's internal parser for rules
and actions instead.

*Work sponsored by SWITCH*

lkiesow committed:
MH-12948, Directly read XACML files

The XACML authorization service passed configuration files to a JBoss
library to determine if a specific action is allowed for a user given an
XACML file. The configuration itself contained file locations specifying
where the library could load the XACML file. This caused Opencast to
repeatedly load the XACML attachments into the workspace, cleaning them
up again afterwards.

This patch now allows Opencast to directly read those files from the
working file repository by using Opencast's internal parser for rules
and actions instead.

smarquard commented:
This looks like a good improvement, +1 especially for getting rid of the oddly-specific use of JBoss libraries.

lkiesow commented:
> +1 especially for getting rid of the oddly-specific use of JBoss libraries.

It's still in other parts of this module though… :-(

lkiesow commented on modules/authorization-xacml/src/main/java/org/opencastproject/authorization/xacml/XACMLAuthorizationService.java:
```diff
@@ -428,88 +418,37 @@ public boolean hasPolicy(MediaPackage mp) {
+    } catch (XACMLParsingException | NotFoundException | IOException e) {
+      return false;
+    }
+    logger.error("acl: {}", acl);
```
this development log needs to be removed.

gregorydlogan commented on modules/authorization-xacml/src/main/java/org/opencastproject/authorization/xacml/XACMLAuthorizationService.java:
```diff
@@ -428,88 +418,36 @@ public boolean hasPolicy(MediaPackage mp) {
+    try {
+      acl = XACMLUtils.parseXacml(workspace.read(attachment.getURI()));
+    } catch (XACMLParsingException | NotFoundException | IOException e) {
+      return false;
```
Do we want to log this error?

gregorydlogan commented on modules/authorization-xacml/src/main/java/org/opencastproject/authorization/xacml/XACMLAuthorizationService.java:
```diff
@@ -428,88 +418,36 @@ public boolean hasPolicy(MediaPackage mp) {
-            return pdp.evaluate(requestCtx).getDecision() == XACMLConstants.DECISION_PERMIT;
+    Option<Attachment> xacml = getXacmlAttachment(mp);
+    if (xacml.isNone()) {
+      return false;
```
I'd be tempted to pop a debug or trace log here.

gregorydlogan reviewed (approved):
Approved, but I would add a couple of logging lines to make life easier.

lkiesow committed:
MH-12948, Log XACML evaluation

This patch adds additional log messages for the evaluation of XACML
files in Opencast.

lkiesow commented:
I've added additional logs to be able to trace the evaluation of XACML files in Opencast. I'm not entirely sure how the warn level log will turn out on a production system. It might spam the logs (or not ;-) but we can evaluate that over the next month. It will be pretty obvious if it happens and will do no harm.
